### PR TITLE
feat: [DHIS2-10943] enrollemnt actions

### PR DESCRIPTION
**Pull request by simonadomnisoru**

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetEnrollment/Actions/Cancel/Cancel.component.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import type { Props } from './cancel.types';
+import { plainStatus } from '../../constants/status.const';
+
+export const Cancel = ({ enrollment, updateAction }: Props) =>
+    (enrollment.status === plainStatus.CANCELLED ? (
+        <MenuItem
+            dense
+            dataTest="widget-enrollment-actions-reactivate"
+            onClick={async () =>
+                updateAction({
```
[the remark this answers is not in the document]
Comment (simonadomnisoru) [reply src/core_modules/capture-core/components/WidgetEnrollment/Actions/Cancel/Cancel.component.js:14]: Hey, 
It seems I forgot to remove the `async`. It is deleted now. Thank you for your feedback!

**Review by JoakimSM — COMMENTED** [review]
Overall looks good @simonadomnisoru 👍  I like the way you structured your code.

There is one aspect I would like us to discuss related to the api requests. My suggestion would be that we show an intermediate status while the request is processing (completing.., cancelling...) and also disable the dropdown during this time. Currently if you're on bad connection, it could potentially take some time until you get any kind of feedback and you're able to click multiple of the actions, potentially giving you an unexpected result. For example, doing complete first then "mark for follow-up" will result in an enrollment marked for follow up, but not completed. There are other ways to deal with this of course, but this might be easiest solution.

Please share your thoughts @Bekkalizer @simonadomnisoru

**Comment by Bekkalizer** [comment]
> Overall looks good @simonadomnisoru 👍 I like the way you structured your code.
> 
> There is one aspect I would like us to discuss related to the api requests. My suggestion would be that we show an intermediate status while the request is processing (completing.., cancelling...) and also disable the dropdown during this time. Currently if you're on bad connection, it could potentially take some time until you get any kind of feedback and you're able to click multiple of the actions, potentially giving you an unexpected result. For example, doing complete first then "mark for follow-up" will result in an enrollment marked for follow up, but not completed. There are other ways to deal with this of course, but this might be easiest solution.
> 
> Please share your thoughts @Bekkalizer @simonadomnisoru

I think this seems like a good solution. The behavior will potentially be a bit strange if we manage to send two update calls at the same time, or almost at the same time: One of the calls would "win" and the changes from the other call would be lost I think 🤔

**Comment by simonadomnisoru** [comment]
> Overall looks good @simonadomnisoru 👍 I like the way you structured your code.
> 
> There is one aspect I would like us to discuss related to the api requests. My suggestion would be that we show an intermediate status while the request is processing (completing.., cancelling...) and also disable the dropdown during this time. Currently if you're on bad connection, it could potentially take some time until you get any kind of feedback and you're able to click multiple of the actions, potentially giving you an unexpected result. For example, doing complete first then "mark for follow-up" will result in an enrollment marked for follow up, but not completed. There are other ways to deal with this of course, but this might be easiest solution.
> 
> Please share your thoughts @Bekkalizer @simonadomnisoru

I agree with your suggestions. I will disable the dropdown while the request is being processed. I will also show the loading widget and a message informing the user about the state. I will look something like this. 
Screenshot 2021-06-03 at 10 31 30
Will this improvement address all of your concerns? Thanks!

**Review by JoakimSM — APPROVED** [review]
Perfect! 👍

**Pull request merged**